core/src/main/java/com/datastax/oss/driver/api/core/data/GettableByIndex.java

```diff
@@ -160,9 +170,12 @@ default Object getObject(int i) {
   default boolean getBoolean(int i) {
     DataType cqlType = getType(i);
     TypeCodec<Boolean> codec = codecRegistry().codecFor(cqlType, Boolean.class);
-    return (codec instanceof PrimitiveBooleanCodec)
-        ? ((PrimitiveBooleanCodec) codec).decodePrimitive(getBytesUnsafe(i), protocolVersion())
-        : get(i, codec);
+    if (codec instanceof PrimitiveBooleanCodec) {
```

Comment:
why change this?  It doesn't seem to change the behavior (nulls get changed to false in boolean conversion)

Reply:
There was a risk of NPE in the previous version:
```
    return (codec instanceof PrimitiveByteCodec)
        ? ((PrimitiveByteCodec) codec).decodePrimitive(getBytesUnsafe(i), protocolVersion())
        : get(i, codec);
```
The unboxing of `get(i, codec)` will throw NPE if that method returns `null` (which can happen depending on which codec has been selected).